### PR TITLE
Fix saved search issues (release-2.1)

### PR DESCRIPTION
**Pull request by srfarley**
- [ ] @joeferner
- [ ] @mwizeman @dsingley @EvanOxfeld 
- [x] @joeybrk372 @sfeng88 @rygim @jharwig 

===
* Fix issues so saved searches can switch between global and private with the appropriate user privilege
* Add more tests for global saved searches
* Return all privileges using Privilege.ALL
* Add SEARCH_SAVE_GLOBAL to UserPropertyPrivilegeRepository.defaultPrivileges config property

### CODE REVIEW

File: web/client-api/src/main/java/org/visallo/web/clientapi/model/Privilege.java [hunk 1]
```diff
@@ -15,7 +15,19 @@
...
+                READ, COMMENT, COMMENT_EDIT_ANY, COMMENT_DELETE_ANY, SEARCH_SAVE_GLOBAL, EDIT, PUBLISH, ADMIN
+        };
+        Set<Privilege> allPrivileges = new HashSet<Privilege>(allNames.length);
+        for (String name : allNames) {
+            allPrivileges.add(new Privilege(name));
+        }
+        ALL = Collections.unmodifiableSet(allPrivileges);
+    }
+
+    public static final Set<Privilege> ALL;
```
Comment (joeferner) [inline web/client-api/src/main/java/org/visallo/web/clientapi/model/Privilege.java]: I don't love calling it ALL here since it is only all built in

Comment (srfarley) [inline web/client-api/src/main/java/org/visallo/web/clientapi/model/Privilege.java]: Good point. How about `ALL_BUILT_IN` or `ALL_CORE`?

Comment (joeferner) [inline web/client-api/src/main/java/org/visallo/web/clientapi/model/Privilege.java]: Works for me

**Comment by srfarley** [comment]
@joeferner Renamed to `ALL_BUILT_IN`.

**Pull request merged**